From 4c62dd19fa7d48559acd996bfbbb1d873a07af73 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 22 Sep 2016 15:14:23 +0300
Subject: [PATCH 0516/1345] mvpp2x: set skb checksum level for RX packets

commit  c866bd9de23538612d4d4402ae006210da119354 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- set skb checksum level 1 for RX UDP and TCP packets

Change-Id: I8dc14ac7574829fb59edc74a6a94e97f03fe9766
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33072
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
index 4fad8cc..b2060eb 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
@@ -3454,6 +3454,7 @@ void mv_pp2x_rx_csum(struct mv_pp2x_port *port, u32 status,
 		     (status & MVPP2_RXD_L4_TCP)) &&
 		     (status & MVPP2_RXD_L4_CSUM_OK)) {
 			skb->ip_summed = CHECKSUM_UNNECESSARY;
+			skb->csum_level = 1;
 			return;
 		}
 
-- 
1.7.9.5

